dms/models/dms_security_mixin.py

```diff
@@ -287,8 +311,30 @@ def filter_access_groups(self, operation):
             )
             self.env.cr.execute(sql_query, [self.env.user.id])
             ids_with_access += list(map(lambda val: val[0], self.env.cr.fetchall()))
+            # Add related to record_ref_mail_follower_ids
+            if self._table == "dms_directory":
+                ids_with_access += self._get_directory_ids_with_record_ref()
         return self & self.browse(ids_with_access)
 
+    def _get_directory_ids_with_record_ref(self):
+        ids = []
+        directories = (
+            self.env["dms.directory"]
+            .sudo()
+            .search(
+                [
+                    ("storage_id.inherit_access_from_parent_record", "=", True),
+                    ("res_model", "!=", False),
+                    ("res_id", "!=", 0),
+                ]
+            )
+        )
+        for directory in directories:
+            record = self.env[directory.res_model].browse(directory.res_id)
+            if record.check_access_rights("read") and directory.id not in ids:
```

Comment:
AFAICS you have another bug here.

`check_access_rights()` is an `@api.model` method. This means that it's meant to be executed once per model, not per record.

OTOH, you also have `check_access_rule()`, which is `@api.multi` and should be checked per record, but I don't see it being used anywhere.

Reply:
I have added some tests to check it and `check_access_rights` and `check_access_rule` don't work fine because allways return True although in some cases need to return False (remember in some case some records it's not accessible, private address for example). For these reasons i have used `record.read()`